backend/src/stwo/circuit_builder.rs

```diff
@@ -95,21 +109,34 @@ impl<T: FieldElement> FrameworkEval for PowdrEval<T> {
             .filter_map(|id| id.try_into().ok())
             .collect::<Vec<_>>();
 
-        
-            polynomial_identities.iter().for_each(|id| {
-                let expr =
-                    to_stwo_expression(&self.witness_columns, &id.expression, &witness_eval, &eval);
-                eval.add_constraint(expr);
-            });
-        
+        polynomial_identities.iter().for_each(|id| {
+            let expr =
+                to_stwo_expression(&self.witness_columns, &id.expression, &witness_eval, &eval);
+            eval.add_constraint(expr);
+        });
 
-        for id in self.analyzed.identities_with_inlined_intermediate_polynomials() {
+        for id in self
+            .analyzed
+            .identities_with_inlined_intermediate_polynomials()
+        {
             match id {
                 // Already handled above
                 Identity::Polynomial(..) => {}
```

Comment:
It would be clearer to treat the polynomial identities here, instead of extracting them before and ignoring them here.

Reply:
done